### DISCUSSION TITLE
Before nuclear genome assembly, is it recommended to remove reads of organelles (mitochondria, chloroplasts) and plasmids?

### DISCUSSION THREAD

**Discussion by maruiqi0710** [ablab/spades#1139] [Q&A] [answered]
Which of the following pipelines is recommended? Or can you provide a new recommended pipeline? 

pipeline 1
1. Use plasmidspades.py and organelle assembly software to assemble the genomes of organelles (mitochondria, chloroplasts) and plasmids.
2. Remove **reads** of plasmids and organelles from clean fastq files based on alignment.
3. Run spades.py to assemble the nuclear genome.

pipeline 2
1. Run spades.py to assemble the nuclear genome.
2. Use plasmidspades.py and organelle assembly software to assemble the genomes of organelles (mitochondria, chloroplasts) and plasmids.
3. Remove **contigs** of plasmids and organelles from the nuclear genome based on alignment.

**Comment by asl** [comment, marked as answer]
That's hard question :) And there is no definite recipe / correct answer as it depends on many factors. For example, if you'd have some common fragments between nuclear genome and say, plasmid, e.g. due to some prophages or other factors and removing plasmid reads would certainly fragment (or cause misassemblies!) the nuclear assembly. The opposite is also true though – assembling things separately might improve the assembly results.

But usually joint assembly could be used as a starting point as it contains everything and is (presumably) more correct
